internal/types/document_validation.go

```diff
@@ -62,7 +62,9 @@ func (e *ValidationError) Code() ValidationErrorCode {
 // ValidateData checks if the document represents a valid "data document".
 // It places `_id` field into the fields slice 0 index.
 // If the document is not valid it returns *ValidationError.
-func (d *Document) ValidateData() error {
+// The argument isUpdate is set true when validation is applied
+// upon update operation.
+func (d *Document) ValidateData(isUpdate bool) error {
```

Comment:
I think the initial idea of this function was to keep it as the one that can be used all the time, for any document, regardless of the stage or operation. This is the package `type` and it doesn't really know anything about inserts, updates, deletes, it works with documents themselves.

What if in the future we discover some special cases for `isDelete`, `isUpsert`, `isSomethingElse`? 
My suggestion is not to extend this function but to write a separate function to validate the result of update operation, and then this validate function should be in the place where update operation is happening.

Reply:
Thanks for the tip, let's do the validation in handler which is where update is happening :+1: